### PR TITLE
[DTRA]Ahmad/DTRA-791/Smartcharts preview links

### PR DESCRIPTION
**Pull request by ahmadtaimoor-deriv**
This PR creates the preview links for the smartcharts

### CODE REVIEW

**Review by review-deriv — COMMENTED** [review]
PR description is too short. You can use this format:

```
# Description
This is a summary of the changes. This should describe why do we need this change.
Whys and Hows come here.

# Cards
* https://clickup.com/12345
* https://clickup.com/12346

```

**Review by review-deriv — COMMENTED** [review]
PR description is too short. You can use this format:

```
# Description
This is a summary of the changes. This should describe why do we need this change.
Whys and Hows come here.

# Cards
* https://clickup.com/12345
* https://clickup.com/12346

```

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#  Configuration of code ownership and review approvals for the binary-com/deriv-app repo.
+#
+#  More info: https://help.github.com/articles/about-codeowners/
+#
+
+* @binary-com/write-admin-smartcharts
```
Comment (balakrishna-deriv) [inline .github/CODEOWNERS:6]:
```suggestion
* @binary-com/write-admin-smartcharts

```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: npm
```
Comment (balakrishna-deriv) [inline .github/dependabot.yml:3]: it's missing github actions ecosystem

File: .github/workflows/generate_preview_link.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        runs-on: Ubuntu-latest
+        permissions:
+          checks: write
+          pull-requests: write
+        steps:
+          - name: Verify user
+            uses: "deriv-com/shared-actions/.github/actions/verify_user_in_organization@v1"
+            with:
+              username: ${{github.event.pull_request.user.login}}
+              token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```
Comment (balakrishna-deriv) [inline .github/workflows/generate_preview_link.yml:22]:
```suggestion
              token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}    
              
```

File: .github/workflows/generate_preview_link.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                  repository: ${{github.event.pull_request.user.login}}/deriv-app
+                  path: deriv-app
+                  ref: ${{github.head_ref}}
+                  alternate_repository: 'binary-com/deriv-app'
+                  alternate_ref: master
+
+          - name: Custom flutter-chart
+            uses: 'binary-com/SmartCharts/.github/actions/checkout@master'
+            with:
+                  repository: ${{github.event.pull_request.user.login}}/flutter-chartz
```
Comment (balakrishna-deriv) [inline .github/workflows/generate_preview_link.yml:51]:
```suggestion
                  repository: ${{github.event.pull_request.user.login}}/flutter-chart
```

File: .github/workflows/generate_preview_link.yml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                  output_dir: deriv-app/packages/core/dist
+
+          - name: 'Generate preview link comment'
+            if: always() && steps.post_preview_build_comment.outcome == 'success'
+            uses: 'deriv-com/shared-actions/.github/actions/post_preview_link_comment@v1'
+            with:
+                  issue_number: ${{github.event.number}}
+                  check_run_id: ${{steps.post_preview_build_comment.outputs.check_run_id}}
+                  preview_url: ${{steps.publish_to_pages_branch.outputs.cf_pages_url}}
+                  status: ${{job.status}}
```
Comment (balakrishna-deriv) [inline .github/workflows/generate_preview_link.yml:125]:
```suggestion
                  status: ${{job.status}}

```

File: .github/dependabot.yml (hunk 2 above)
Comment (ahmadtaimoor-deriv) [reply .github/dependabot.yml:3]: Added , thanks

**Review by balakrishna-deriv — APPROVED** [review]

**Pull request merged**